openff/pablo/_additional_definitions.py

```diff
@@ -130,14 +131,14 @@ def _apply_resdef_to_graph(
     pdb_graph: Graph[int, tuple[int, int]],
     atoms: dict[int, AtomDefinition | None],
     bonds: dict[tuple[int, int], BondDefinition | None],
-) -> "AdditionalDefMatch | None":
-    """Find a way to match this additional residue definition to the graph
+) -> "list[AdditionalDefMatch]":
+    """Find all ways to match this additional residue definition to the graph
 
     Raises
     ======
     ValueError
```

Comment:
(not blocking) This is now the only error type explicitly raised in this function. 
```suggestion
    AmbiguousMatchError
```